### ISSUE TITLE
eclipse launch script always launches main workspace

### ISSUE THREAD

**Issue by hohwille** [devonfw/devon-ide#229]
1. create a new sub-folder in workspaces (`mkdir workspaces/test`)
2. create launch script for it (`cd workspaces/test && devon eclipse create-script`)
3. double click the new lauch script (`eclipse-test.bat`)
4. Eclipse opens in the `main` workspace instead of the new `test` workspace.

**Comment by hohwille** [comment]
Added some debugging output:
```
C:\projects\cool-project\workspaces\test C:\Program Files\Git\bin\bash.exe -c source ~/.devon/devon eclipse
```
Until here (`devon.bat` all goes right). I assume that when passing over to git bash here the current working dirctory is lost for whatever reason...

**Comment by hohwille** [comment]
Adding debug output to `devon` bash script:
```
/c/projects/cool-project/workspaces/test
running command eclipse in /c/projects/cool-project/workspaces/test
```
So until the `eclipse` commandlet is called again everything is fine.

**Comment by hohwille** [comment]
OK. So it seems that the devon bash script is invoked in to correct location and also properly detects and sets the `WORKSPACE` variable.
However, that variable is not exported (see #5). As the `devon` bash script is then calling the `eclipse` commandlet it is creating a sub-shell. This only inherits exported variables so `WORKSPACE` variable is lost.
The `environment-project` script then initialized the `WORKSPACE` to `main` as a fallback:
https://github.com/devonfw/devon-ide/blob/5a3fb640fb9942149e6110453e82da02e810395a/scripts/src/main/resources/scripts/environment-project#L59

**Comment by hohwille** [comment]
To fix this there are two options:
A. Export `WORKSPACE`
B. move the workspace detection from `devon[.bat]` scripts to `environment-project[.bat]` scripts.

A. is easy but bad regarding #5 
B. is a bigger change with the risk to break existing functionality.
However, I would tend to B.
I anyhow redesigned the `devon-ide` such that `environment-project` is called once:
* either directly from `devon` script if no args are provided to setup env
* or otherwise from a commandlet when `functions` are sourced that themselves source `environment-project`.

(Before this was partially done twice wasting performance to load arbitrary `devon.properties`).
This new design was introduced exactly to avoid the need to export variables for passing from `devon` env init to commandlets.

**Comment by hohwille** [comment]
Problem with B. is that the detection for `DEVON_IDE_HOME` is required in `devon` script and cannot be avoided. This detection also detects the workspace.
If we move the workspace detection to `environment-project` then we have to do the directory traversal again there also duplicating code.
So for now I probably better go the easy way and export `PATH`.

**Comment by hohwille** [comment]
Fixed with PR #230
